matthewhughes934/isort-fork
# Update deps

- Drop some unused dependencies

    * `ipython` was removed with 49b41a5112ca3e8305dc4836e871ad9ab7f66792
    * `smmap2` looking at the history with `git log -G smmap --name-only
      --oneline` suggests this was only ever defined in
      `poetry.lock`/`pyproject.toml`. Given this is a library and not
      providing something extra like e.g. a `flake8` plugin it should be
      safe to remove

- Drop `py` dependency

    This package is in maintenance only mode and only had two uses. Replace
    their use with the `tmp_path` fixture as suggested in the docs[1]

    Link: https://docs.pytest.org/en/8.3.x/reference/reference.html#tmpdir [1]

- Require Python >=3.8.1

    The motivation here is to bring in a new version of `flake8`, which has
    similar version constraints[1]. Python 3.8.0 is very old (and Python 3.8
    is EoL anyway)

    Link: https://github.com/PyCQA/flake8/commit/aa002ee4ed7fcb4c6ffdec29b4dc122244638c64 [1]

- Update `pytest`

    There was one new error:

        TypeError: exceptions must be derived from Warning, not <class 'NoneType'>

    Fix this following the docs[1]

    Link: https://docs.pytest.org/en/8.3.x/how-to/capture-warnings.html#additional-use-cases-of-warnings-in-tests [1]

- Update `black`

    Also some fixes for 5b74abaf27d43fd4502fd1993e0991c7f347045d

    * loosen the version constraint
    * restore the `--check` flag to the lint script

- Update `flake8`

    This required fixes an error in the config (I also updated the relevant
    link, the old one would 404):

        ValueError: Error code '#' supplied to 'extend-ignore' option does not match '^[A-Z]{1,3}[0-9]{0,3}$'

    And some errors in the code:

        isort/literal.py:59:8: E721 do not compare types, for exact checks use `is` / `is not`, for instance checks use `isinstance()`
        tests/unit/test_isort.py:13:1: F401 'typing.Dict' imported but unused

    The second test conflicted with `mypy`, since `Dict` was only used in
    commented annotations, to make them both happy move one of these
    annotations out of a comment (in the future we should probably move all
    such annotations out of comments)

- Update `hypothesis`

    This required addressing some errors from `mypy`:

        tests/integration/test_setting_combinations.py:20: error: "SearchStrategy" expects no type arguments, but 1 given  [type-arg]
        tests/integration/test_hypothesmith.py:36: error: "SearchStrategy" expects no type arguments, but 1 given  [type-arg]

- Update remaining dependencies

    A bunch of updates that required no additional changes:

    * Update `setuptools`
    * Update `bandit`
    * Update `httpx`
    * Update `pytest-mock`
    * Update `pep8-naming`
    * Update `pip_api`
    * Update `pip`
    * Update `safety`
    * Update `types-toml`
    * Update `vulture`
    * Update `coverage`
    * Update `types-colorama`
